### PR TITLE
content: add CEOI 09-Harbingers

### PR DESCRIPTION
**Pull request by thecodingwizard**
lol I haven't actually solved this problem yet so I'm not sure if the solution / the difficulty is correct XD

### CODE REVIEW

**Comment by dolphingarlic** [comment]
I think it should be at least "Normal". I didn't find the CHT on trees thing immediately obvious, and the trick isn't mentioned anywhere else in the module.

File: content/5_Plat/CHT.mdx [hunk 1]
```diff
@@ -40,6 +40,15 @@ export const problems = {
...
 			''
 		),
+		new Problem(
+			'CEOI',
+			'2009 - Harbingers',
+			'https://oj.uz/problem/view/CEOI09_harbingers',
+			'Normal',
+			false,
+			['DP', 'convex'],
+			'CHT on a tree -- dfs on the tree and keep track of which lines you remove so you can "restore" the deque of lines when leaving a node'
```
Comment (dolphingarlic) [inline content/5_Plat/CHT.mdx:50]: Also, the official editorial is https://infosb.net/ceoi2009/tasks/solutions.doc (Mostafa has pdf versions on his GitHub as well)

**Comment by bqi343** [comment]
I always thought you were just supposed to use line container and small to large merging :P

**Comment by nhtrnm** [comment]
> I always thought you were just supposed to use line container and small to large merging :P

That is only true is the answer for the current node depends on the CHT of the nodes in the subtree. In this problem, it's CHT of the ancestor nodes.

Or in other words, if the DP is from leaves up to root then you are correct, it's just line container and small to large merging. If it's the other way round then persistent CHT. And in this case, it can nicely be implemented in a stack.

**Comment by thecodingwizard** [comment]
uhh does it need to be removed from advanced - linecontainer then (?)

**Comment by nhtrnm** [comment]
> uhh does it need to be removed from advanced - linecontainer then (?)

My apologies, I just stumbled across your repo and while solving this problem so I don't know what levels there are. But this problem certainly does not require line container (presumably you are talking about the KACTL one). But it's also not a standard CHT problem as in you can't just come up with the recurrence and voila you are done. I'd say it's hard but not too complex in terms of the data structures that one needs to understand to solve this.

**Comment by thecodingwizard** [comment]
ah ok perfect, thanks!! :)

**Comment by nhtrnm** [comment]
Do you collect solutions for problems? Asking because I could give you mine.

**Comment by dolphingarlic** [comment]
We do collect editorials for [problems in the guide](https://usaco.guide/problems/), but since this particular problem already has an official editorial, you don't need to write one for us. You're more than welcome to give us your solutions to problems that don't currently have editorials though (e.g. many of the POI problems)!

**Comment by nhtrnm** [comment]
Gotcha!

**Comment by bqi343** [comment]
Ok, it's probably because I never implemented this :P

File: content/5_Plat/CHT.mdx [hunk 2]
```diff
@@ -40,6 +40,15 @@ export const problems = {
...
 			''
 		),
+		new Problem(
+			'CEOI',
+			'2009 - Harbingers',
+			'https://oj.uz/problem/view/CEOI09_harbingers',
+			'Normal',
+			false,
+			['DP', 'convex'],
+			'CHT on a tree -- dfs on the tree and keep track of which lines you remove so you can "restore" the deque of lines when leaving a node. Editorial: https://infosb.net/ceoi2009/tasks/solutions.doc'
```
Comment (dolphingarlic) [inline content/5_Plat/CHT.mdx:50]:
```suggestion
			'https://infosb.net/ceoi2009/tasks/solutions.doc'
```

**Pull request merged**